Core/src/V2/Iam.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\V2;
+
+use Google\ApiCore\Serializer;
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Core\Iam\PolicyBuilder;
+use Google\Cloud\Iam\V1\Policy;
+use Google\Cloud\Core\RequestHandler;
+use Google\Cloud\Iam\V1\GetIamPolicyRequest;
+use Google\Cloud\Iam\V1\GetPolicyOptions;
+use Google\Cloud\Iam\V1\SetIamPolicyRequest;
+use Google\Cloud\Iam\V1\TestIamPermissionsRequest;
+
+/**
+ * IAM Manager
+ *
+ * This class is not meant to be used directly. It should be accessed
+ * through other objects which support IAM.
+ *
+ * Policies can be created using the {@see Google\Cloud\Core\Iam\PolicyBuilder}
+ * to help ensure their validity.
+ *
+ * Example:
+ * ```
+ * // IAM policies are obtained via resources which implement IAM.
+ * // In this example, we'll use PubSub topics to demonstrate
+ * // how IAM policies are managed.
+ *
+ * use Google\Cloud\PubSub\PubSubClient;
+ *
+ * $pubsub = new PubSubClient();
+ * $topic = $pubsub->topic('my-new-topic');
+ *
+ * $iam = $topic->iam();
+ * ```
+ *
+ * @internal
+ */
+class Iam
+{
+    use ArrayTrait;
+
+    /**
+     * @var RequestHandler
+     */
+    private $requestHandler;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    private $gapic;
+
+    /**
+     * @var string
+     */
+    private $resource;
+
+    /**
+     * @var array
+     */
+    private $policy;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @param RequestHandler $requestHandler
+     * @param Serializer $serializer The serializer instance to encode/decode messages.
+     * @param string $resource
+     * @access private
+     */
+    public function __construct(
+        RequestHandler $requestHandler,
```

Comment:
Since the GAPIC clients are part of the `RequestHandler` class, couldn't we just put them in there? 

E.g. https://github.com/googleapis/google-cloud-php/pull/6871/files#diff-a5f7e5f451db2d7ff64cee25bb4307c686cb556e4789e98eeec1c4f5008c3053R61

It's strange to have a local `$gapic` object, but also have a `RequestHandler` with a `$gapics` property.

Also, as we mentioned, we should try to not use the term "GAPIC" anywhere. Terms such as `Client`, `Service`, etc would be better.

Reply:
Ahh I see that this is a classname and not a client class. Part of the reason is that the variable should have a better name (e.g. `$clientClass`, as discussed in https://github.com/googleapis/google-cloud-php/pull/6871) and another reason is that the property does not contain a typehint.